x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -254,9 +256,22 @@ export class ObservabilityAIAssistantClient {
           }
 
           if (isAssistantMessageWithFunctionRequest) {
-            const span = apm.startSpan(
-              `execute_function ${lastMessage.message.function_call!.name}`
-            );
+            const functionCallName = lastMessage.message.function_call!.name;
+
+            if (functionClient.hasAction(functionCallName)) {
+              this.dependencies.logger.debug(`Executing client-side action: ${functionCallName}`);
+
+              functionClient.validate(
```

Comment:
This throws an error if it fails? And if it fails, then what happens with the `subscriber`? Does it not complete?

Reply:
the error falls through and eventually subscriber.error is called. but that shouldn't actually happen here, we should inject a function error message and send it back to the LLM. will fix and add a test, thanks.